app/src/androidTest/java/ch/epfl/culturequest/ui/ProfileFragmentTest.java

```diff
@@ -93,32 +98,25 @@ public void settingButtonWorks() {
     }
 
     @Test
-    public void deleteButtonWorks() {
-        Post post = new Post("abc", "cT93LtGk2dT9Jvg46pOpbBP69Kx1",
-                "https://firebasestorage.googleapis.com/v0/b/culturequest.appspot.com/o/0000598561_OG.jpeg?alt=media&token=503f241d-cebf-4050-8897-4cbb7595e0b8",
-                "Piece of Art", 0, 0, new ArrayList<>());
-
-        Database.uploadPost(post).whenComplete((a,b) -> {
-            assertEquals(1, Objects.requireNonNull(Database.getPosts(Profile.getActiveProfile().getUid(),1,0).join()).size());
-
-            //long click on the first picture should open an alert dialog
-            onView(withId(R.id.pictureGrid)).perform(RecyclerViewActions.actionOnItemAtPosition(0, ViewActions.longClick()));
-            // should open an alert dialog
-            onView(withText("Are you sure you want to delete this post?")).check(matches(isDisplayed()));
-            onView(withText("No")).perform(click());
-            onView(withText("Are you sure you want to delete this post?")).check(doesNotExist());
-            assertEquals(1, Objects.requireNonNull(Database.getPosts(Profile.getActiveProfile().getUid()).join()).size());
-
-            onView(withId(R.id.pictureGrid)).perform(RecyclerViewActions.actionOnItemAtPosition(0, ViewActions.longClick()));
-            onView(withText("Are you sure you want to delete this post?")).check(matches(isDisplayed()));
-            onView(withText("Yes")).perform(click());
-            onView(withText("Are you sure you want to delete this post?")).check(doesNotExist());
-            assertEquals(0, Objects.requireNonNull(Database.getPosts(Profile.getActiveProfile().getUid()).join()).size());
-            onView(withId(R.id.pictureGrid)).check(matches(hasChildCount(0)));
-
-        });
-
-       }
+    public void deleteButtonWorks() throws InterruptedException {
+        Thread.sleep(5000);
+        assertEquals(1, Objects.requireNonNull(Database.getPosts(Profile.getActiveProfile().getUid()).join()).size());
+
+        //long click on the first picture should open an alert dialog
+        onView(withId(R.id.pictureGrid)).perform(RecyclerViewActions.actionOnItemAtPosition(0, ViewActions.longClick()));
+        // should open an alert dialog
+        onView(withText("Are you sure you want to delete this post?")).check(matches(isDisplayed()));
```

Comment:
if you dont put `Database.uploadPost(post).whenComplete((a,b) -> {...` and then put the asserttions in there this test will fail bc of the profileFragment being refreshed by uploading a post. The test will fail with a message "inconsistency detected"

Reply:
I know you put `.join` but as you see in the CI, the test still fails

Reply:
I found the problem, what you did was not the solution, actually by doing that you are just skipping the test because you are doing it asynchronously

After 2 hours of debug I found that the problem came from the "ProfileUtils.postsAdded". When it is equal to 1 due to your "OnResume" logic I don't know why but it breaks eveything ...
So the solution I found was to put it to 0 before each test.

Reply:
ah... well we might need to patch this in the future to resolve this, i checked the pb with this test and it normally doesn't occur with a recycler view instead of a scroll view, we'll check that out

Reply:
Ok no worries we will talk about that tomorrow, I am waiting to see if the CI fails again. It failed due to an OpenAI stuff I run it again.

Reply:
It's working !! :)